src/map/Map.js

```diff
@@ -349,7 +364,7 @@ L.Map = L.Evented.extend({
 			return null;
 		}
 
-		return inside ? zoom : zoom - 1;
+		return inside ? zoom : zoom - snap;

```

Comment:
I've given it a bit of thought and maybe we should do a bisection algorithm. I just don't think it's the right time to do that bit of refactoring.
Also, there is the issue of zoom ramps with irregular steps (e.g. 1, 1.1, 1.5, 2, 2.1, 2.5, 3...). I was giving it some thought, and at some point implemented partial support for a user-provided function to snap a zoom to the nearest step. But for a bisection algorithm to work properly, the user would need to provide `round()`, `floor()` and `ceil()`, so I decided to let the bisection stuff rest for a while.